### ISSUE TITLE
The BMW Connected Drive portal returned an error: UNKNOWN_ERROR

### ISSUE THREAD

**Issue by deluxestyle** [bimmerconnected/bimmer_connected#250]
**Describe the issue**

Not able to connect to the bmw connected portal with the current bimmer_connected version after a restart of home assistant today.

"OSError: The BMW Connected Drive portal returned an error: UNKNOWN_ERROR (received status code 400 and expected 200)."
"Authentication failed. Maybe your password is invalid?"

**Expected behavior**

Login to bmw connected

**Home Assistant version (if used)**

0.117.6

And did it work on a previous version, if so what was the last working version? yes. 0.117.6

**ConnectedDrive website**
https://www.bmw-connecteddrive.de/

If you see any issues with the BMW integration in Home Assistant (or when directly using the library), please make sure to first login to the BMW ConnectedDrive website of your country and check if you can successfully login and query the car status.
- [x] I can still successfully login to the BMW ConnectedDrive website and the car status is available there.
- [x] I have ConnectedDrive enabled for my vehicle.

**Your config.yaml**

```yaml
bmw_connected_drive:
  name_of_car:
    username: USERNAME_BMW_CONNECTED_DRIVE
    password: PASSWORD_BMW_CONNECTED_DRIVE
    region: "rest_of_world"
````

**Number of cars** 1

If you have more than 1 car please mention that here.
- [ ] I have 2 or more cars linked to the ConnectedDrive account.
- [ ] I have a Mini vehicle linked to my account.

**Output of bimmer_connected fingerprint**

If you have run the [bimmer_connected fingerprint](https://github.com/bimmerconnected/bimmer_connected#data-contributions) please share that data here

*NOTE: It might take us a while to respond on GitHub (we only maintain this in our free time), but we will respond.*

**Comment by minimalisticMe** [comment]
They changed the API since yesterday.
I think the revoked the current API/secret, maybe even changed the whole login process?

With a little bit of luck https://github.com/bimmerconnected/bimmer_connected/pull/140 will with this issue.

**Comment by gps1539** [comment]
I also hit this issue today while running 0.7.6. Updating to 0.7.12 did not fix.
I see BMW's android app was updated on 11/05 and while it states 'bug fixes', maybe it was to prep for an API change.

**Comment by rikroe** [comment]
They did change the API auth secret - to the one we already tried in #210.
I am preparing a fix, although I am only able to test this for `rest_of_world` and `north_america`.

**Comment by deluxestyle** [comment]
> They did change the API auth secret - to the one we already tried in #210.
> I am preparing a fix, although I am only able to test this for `rest_of_world` and `north_america`.

Thank you. If you need some tester for rest_of_world I would be able to help

**Comment by rikroe** [comment]
Please test guys :)

Especially would be great if somebody from China would be able to test - although from what I can tell from the response codes, nothing changed there.

**Comment by gps1539** [comment]
Happy to test from North America. What would be the correct way to pull your changes?

**Comment by rikroe** [comment]
If you have a clean Python environment, you could just do:
`pip3 install git+https://github.com/rikroe/bimmer_connected.git@fix-login`

And then the usual `bimmerconnected status ...`.

**Comment by deluxestyle** [comment]
Just copied the files to my homeassistant and it works. 
I'm using "rest_of_world"
thank you

**Comment by rikroe** [comment]
> Just copied the files to my homeassistant and it works.
Yes, the alternative would be to copy `account.py` and `country_selector.py` to your HA instance.

I'll wait for the North America confirmation, then we can release a new version to PyPI and create a PR for Home Assistant.

**Comment by gps1539** [comment]
I updated to fix-login via pip and I can get status. I do see a few 403 errors in the debug output, but it finally authenticates.

DEBUG:bimmer_connected.account:The BMW Connected Drive portal returned an error: UNKNOWN_ERROR (received status code 403 and expected 200).
'
'
DEBUG:bimmer_connected.state:received new data from connected drive

**Comment by rikroe** [comment]
Getting a 403 shouldn't happen. Could provide some more info on which URLs/endpoints you see these?

**Comment by gps1539** [comment]
Here is the debug output, with my VIN and status redacted.

[bc-out.txt](https://github.com/bimmerconnected/bimmer_connected/files/5549389/bc-out.txt)

**Comment by rikroe** [comment]
Thanks! While not expected, this seems to be regarding some other issue. 
I will create a release regarding the login topics and have a look at your 403 issue in the next days.

**Comment by gerard33** [comment]
This is fixed in HA 0.118 which will release 18 November.
Great work @rikroe 👍

**Comment by deluxestyle** [comment]
> This is fixed in HA 0.118 which will release 18 November.
> Great work @rikroe 👍

I'm closing this defect as it was fixed by @rikroe (thanks for the hard word)
